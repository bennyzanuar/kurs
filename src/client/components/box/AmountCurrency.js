import React, { Component } from 'react'
import Typography from '@material-ui/core/Typography';
import Input from '@material-ui/core/Input';

class AmountCurrency extends Component{
    constructor(props){
        super(props)
        this.onChangeAmount = this.onChangeAmount.bind(this);
    }
    onChangeAmount(event){
        event.preventDefault();
        const { amount, changes } = this.props
        changes(event.target.value)
    }
    
    
    render(){
        const { amount } = this.props
        return(
            <>
                <Typography variant="h6">
                    USD - United States Dollars
                </Typography>
                <div>
                    <div className={`width50`}>
                        <Typography variant="h6" component="h6">
                            USD
                        </Typography>
                    </div>
                    <div className={`width50 text-right`}>
                        <Input
                            placeholder="Insert Value"
                            className={``}
                            defaultValue={amount}
                            onChange={ () => {this.onChangeAmount(event)} }
                            inputProps={{
                                style: { textAlign: "right" }
                            }}
                        />
                    </div>
                </div>
            </>
        )
    }
}

export default AmountCurrency