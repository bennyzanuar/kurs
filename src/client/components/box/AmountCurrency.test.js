import React from 'react'
import { shallow } from 'enzyme'
import AmountCurrency from './AmountCurrency'

describe('<AmountCurrency />', () => {
    it('should render amount box', () => {
        const component = shallow(<AmountCurrency />)
        expect(component).toMatchSnapshot()
    })
})