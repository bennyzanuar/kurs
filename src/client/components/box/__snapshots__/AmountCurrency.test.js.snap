// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AmountCurrency /> should render amount box 1`] = `
<Fragment>
  <WithStyles(Typography)
    variant="h6"
  >
    USD - United States Dollars
  </WithStyles(Typography)>
  <div>
    <div
      className="width50"
    >
      <WithStyles(Typography)
        component="h6"
        variant="h6"
      >
        USD
      </WithStyles(Typography)>
    </div>
    <div
      className="width50 text-right"
    >
      <WithStyles(Input)
        className=""
        inputProps={
          Object {
            "style": Object {
              "textAlign": "right",
            },
          }
        }
        onChange={[Function]}
        placeholder="Insert Value"
      />
    </div>
  </div>
</Fragment>
`;
